### ISSUE TITLE
How to do get C API

### ISSUE THREAD

**Issue by Ygg01** [servo/servo#27773]
Had a chat with someone from Hacker News and they had this to say:

> Almost the entire quick start guide is about using Rust. So you know, if I want to embed it, how do I do it?
> 
> If I need a view offscreen rendered to a framebuffer, how do I get it? Can I blit? If so, how? How do I pass events to the browser abstraction?
> 
> Servo talks about none of these things and leaves it up to you, but 110% emphasizes everything possible about Rust.
> 
> I don't care about Rust, I care about embedding web technologies. So how do I do it? They don't explain.
> 
> The front page prioritizes these things, in this order:
> 
>     * How to use Rust
>     * How to contribute to Servo
>     * Servo's blog
>     * How Servo is governed
>     * How to donate to Servo
>     * Contacting those involved with Servo
>     * Downloading Servo
> 
> None of those things point me in the direction of how to embed it.
> 
> Maybe they should spent more time catering to the people they claim to be providing the software for and less to the Rust crowd. 

I think it raises a valid point. Assume I only care about C binding and API, where do I find this documentation and examples? Especially, if this is someone coming from a C world, who knows or cares very little about rust.

**Comment by jdm** [comment]
Yep, it's early days for that. I'm starting with some working examples in https://github.com/jdm/servo-embedding-example.

**Comment by simonschmeisser** [comment]
I'm very new to rust so please bear with me ... it's called capi but I cannot find a c-header? Can those be created automatically from the rust code or am I supposed to recreate all function and type declarations? If so, should I contribute those?

**Comment by jdm** [comment]
It's created in the target directory automatically during the build when you run `./mach build --simpleservo`. There's a `simpleservo.h` that is generated for you.

**Comment by samyak-jain** [comment]
Is there a tracking issue for the status of the embeddable API?

**Comment by jdm** [comment]
No.
